service/common/src/main/java/org/apache/polaris/service/persistence/InMemoryPolarisMetaStoreManagerFactory.java

```diff
@@ -100,10 +96,22 @@ public synchronized Supplier<TransactionalPersistence> getOrCreateSessionSupplie
 
   private void bootstrapRealmsAndPrintCredentials(List<String> realms) {
     RootCredentialsSet rootCredentialsSet = RootCredentialsSet.fromEnvironment();
-    Map<String, PrincipalSecretsResult> results = this.bootstrapRealms(realms, rootCredentialsSet);
+    this.bootstrapRealms(realms, rootCredentialsSet);
     bootstrappedRealms.addAll(realms);
+  }
 
+  @Override
+  public Map<String, PrincipalSecretsResult> bootstrapRealms(
+      Iterable<String> realms, RootCredentialsSet rootCredentialsSet) {
+    Map<String, PrincipalSecretsResult> results = super.bootstrapRealms(realms, rootCredentialsSet);
+
+    Map<String, RootCredentials> presetCredentials = rootCredentialsSet.credentials();
     for (String realmId : realms) {
+      if (presetCredentials.containsKey(realmId)) {
+        // Credentials provided in the runtime env... no need to print
```

Comment:
personally I think this is fine, but this is a small behavior change and could potentially have impact on tests

Reply:
I'll wait another day in case someone else has a concern.